### PR TITLE
exec: access the table_storage_stats need privilege

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: ailinkid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/26128

### What is changed and how it works?

*: What's Changed: add the super privilege control for access table_storage_stats

### Check List 

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

### Release note 

- exec: access the table_storage_stats need super privilege

### CODE REVIEW

**Review by djshow832 — COMMENTED** [review]
The user should be able to access the tables to which he has privileges.

**Comment by AilinKid** [comment]
> The user should be able to access the tables to which he has privileges.

done

File: executor/infoschema_reader_test.go [hunk 1]
```diff
@@ -915,6 +915,29 @@ func (s *testInfoschemaClusterTableSuite) TestTableStorageStats(c *C) {
...
 	c.Assert(len(tk.MustQuery("select TABLE_NAME from information_schema.TABLE_STORAGE_STATS where TABLE_SCHEMA = 'mysql';").Rows()), Equals, 24)
+
+	// More tests about the privileges.
+	tk.MustExec("create user 'testuser'@'localhost'")
+	tk.MustExec("create user 'testuser2'@'localhost'")
+	tk1 := testkit.NewTestKit(c, store)
+	defer tk1.MustExec("drop user 'testuser'@'localhost'")
+	defer tk1.MustExec("drop user 'testuser2'@'localhost'")
+
+	tk.MustExec("grant all privileges on *.* to 'testuser2'@'localhost'")
```
Comment (djshow832) [inline executor/infoschema_reader_test.go:926]: Test only one privilege, such as `select`.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by crazycs520 — APPROVED** [review]

**Pull request merged**